File: 백준_그리디/센서.py
file = open("./백준_그리디/센서tc.txt", "r")

read = file.readline

for _ in range(2) :
	n = int(read())
	k = int(read())
	sensors = list(map(int, read().split()))
	answer = int(read())
	read()
	
	print(n, k, sensors, answer)

	sensors.sort()

	# print(sensors)

	# 각 센서 사이 거리를 구함
	distances = [sensors[i] - sensors[i-1] for i in range(1, n)]
	# print(distances)

	# 센서끼리 거리 중 k - 1개만 제거하면 됨
	distances.sort()

	# print(distances[:len(distances) - (k-1)])

	print(sum(distances[:len(distances) - (k-1)]))
	print()

file.close()

# 백준 제출용
n = int(input())
k = int(input())

sensors = list(map(int, input().split()))
sensors.sort()

distances = [sensors[i] - sensors[i-1] for i in range(1, n)]
distances.sort()

print(sum(distances[:len(distances) - (k-1)]))

File: src/백준/문제집/그리디/센서.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class 센서 {
    /** answer
     * 5
     * 7
     */
    static int n;
    static int k;
    static long[] sensors;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(
                new FileReader(
                        "D:\\git\\IdeaProjects\\Daily_Algo_Exercise\\백준\\문제집\\그리디\\센서.txt"
                )
        );
//        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        k = Integer.parseInt(br.readLine());

        sensors = Arrays.stream(br.readLine().split(" ")).mapToLong(Long::parseLong).toArray();

        Arrays.sort(sensors);

//        System.out.println(Arrays.toString(sensors));

        long[] diff = new long[n-1];

        for (int i = 0; i < n-1; i ++) {
            diff[i] = sensors[i+1] - sensors[i];
        }

        Arrays.sort(diff);

        long result = 0;
        for (int i = 0; i < n-k; i++) {
            result += diff[i];
        }
        System.out.println(result);
    }
}
